### PR TITLE
Zone related email validation

### PR DESCRIPTION
**Pull request by ssranjani06**
Fixes 
https://github.com/vinyldns/vinyldns/issues/1237
https://github.com/vinyldns/vinyldns/issues/1241

Changes in this pull request:

Email Validation for Zone with unit tests

Write functional tests

Todo:

Dropdown displaying valid domains

Just like dotted hosts make the number of dots allowed after @ in an email to be configurable

### CODE REVIEW

File: modules/core/src/main/scala/vinyldns/core/Messages.scala [hunk 1]
```diff
@@ -85,4 +85,6 @@ object Messages {
   val EmailValidationErrorMsg = "Please enter a valid Email ID. Valid domains should end with"
 
   val InvalidEmailValidationErrorMsg = "Please enter a valid Email ID."
```
Comment (nspadaccino) [inline modules/core/src/main/scala/vinyldns/core/Messages.scala:87]: We can remove "ID" from this message

File: modules/core/src/main/scala/vinyldns/core/Messages.scala [hunk 2]
```diff
@@ -85,4 +85,6 @@ object Messages {
   val EmailValidationErrorMsg = "Please enter a valid Email ID. Valid domains should end with"
```
Comment (nspadaccino) [inline modules/core/src/main/scala/vinyldns/core/Messages.scala:85]: We can remove "ID" from this message

**Comment by nspadaccino** [comment]
modules/docs/src/main/mdoc/operator/config-api.md should be updated with the information about the number of dots config, both the Email Domain Configuration and Full Example Config sections. Include what happens if left empty

File: modules/core/src/main/scala/vinyldns/core/Messages.scala (hunk 1 above)
Comment (ssranjani06) [reply modules/core/src/main/scala/vinyldns/core/Messages.scala]: > We can remove "ID" from this message

Done

File: modules/core/src/main/scala/vinyldns/core/Messages.scala [hunk 3]
```diff
@@ -85,4 +85,6 @@ object Messages {
...
 
   val InvalidEmailValidationErrorMsg = "Please enter a valid Email ID."
+
+  val DotsValidationErrorMsg = "Please enter a valid Email ID. Number of dots allowed after @ is"
```
[the remark this answers is not in the document]
Comment (ssranjani06) [reply modules/core/src/main/scala/vinyldns/core/Messages.scala]: > We can remove "ID" from this message

done

File: modules/portal/public/css/vinyldns.css [hunk 4]
```diff
@@ -538,4 +538,16 @@ input[type="file"] {
...
+
+.blink {
+    animation: blinker 1.5s linear infinite;
+    color: red;
+    font-family: sans-serif;
+}
+@keyframes blinker {
+    50% {
+        opacity: 0;
+    }
```
Comment (Aravindh-Raju) [inline modules/portal/public/css/vinyldns.css:550]: We can remove these CSS which is not used, maybe later in a separate PR

File: modules/portal/public/lib/controllers/controller.manageZones.js [hunk 5]
```diff
@@ -325,6 +353,8 @@ angular.module('controller.manageZones', ['angular-cron-jobs'])
             $scope.currentManageZoneState = $scope.manageZoneState.UPDATE;
             $scope.refreshAclRuleDisplay();
             $scope.refreshZoneChange();
+            $scope.validDomains();
+           /* $scope.CollapsedModalUpdateZone();*/
```
Comment (Aravindh-Raju) [inline modules/portal/public/lib/controllers/controller.manageZones.js:388]: We can remove this too.

File: modules/portal/public/templates/zoneconnection-modal.html [hunk 6]
```diff
@@ -17,6 +17,24 @@
...
+                            <span class="badge" id="validDomainsConnectZone">Valid Domains for Email <span class="fa fa-chevron-down"></span></span>
+                        </a>
+                    </h5>
+                </div>
+                <div class="collapse" id="collapseConnectZoneInstruction">
+                    <div class="well">
+                        <p>
+                            List Of Valid Email Domains:
+                        <li ng-repeat="validDomains in validEmailDomains">
+                            {{validDomains}}
```
Comment (Aravindh-Raju) [inline modules/portal/public/templates/zoneconnection-modal.html:32]: I think we shouldn't display this when the valid domains config list is empty as we discussed before.
Screenshot 2023-04-11 at 6 13 54 PM

**Review by Aravindh-Raju — APPROVED** [review]
LGTM!

**Review by nspadaccino — APPROVED** [review]

**Review by arpit4ever — APPROVED** [review]
Verified locally with various scenarios. Looks good.

**Pull request merged**